base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -647,9 +653,15 @@ where T: BlockchainBackend
     /// is reorganised if necessary.
     fn handle_possible_reorg(&self, block: Block) -> Result<BlockAddResult, ChainStorageError> {
         let metadata = self.get_metadata()?;
-        let horizon_block_height = metadata
-            .horizon_block()
-            .ok_or(ChainStorageError::InvalidQuery("Blockchain database is empty".into()))?;
+        let db_height = match metadata.height_of_longest_chain {
```

Comment:
```suggestion
        let db_height = metadata.height_of_longest_chain.ok_or(ChainStorageError::InvalidQuery("Cannot retrieve block. Blockchain DB is empty".into()))?;
```
Then ln 657 to ln 663 can be removed.

Reply:
Yup, nice.

Reply:
BTW - did you know that you can select multiple lines for comment now?

Reply:
I used that pattern in 2 other places as well.

Reply:
I didn't know, will make use of it in the future.